securesystemslib/rsa_keys.py

```diff
@@ -326,11 +326,10 @@ def create_rsa_signature(private_key, data, scheme='rsassa-pss-sha256'):
     if scheme.startswith('rsassa-pss'):
       # Generate an RSSA-PSS signature.  Raise
       # 'securesystemslib.exceptions.CryptoError' for any of the expected
-      # exceptions raised by pyca/cryptography. 'salt_length' is set to
-      # the maximum length available.
+      # exceptions raised by pyca/cryptography.
       signature = private_key_object.sign(
           data, padding.PSS(mgf=padding.MGF1(digest_obj.algorithm),
-          salt_length=padding.PSS.MAX_LENGTH), digest_obj.algorithm)
+          salt_length=padding.PSS.DIGEST_LENGTH), digest_obj.algorithm)
```

Comment:
The [original code](https://github.com/secure-systems-lab/securesystemslib/pull/422/files#diff-3d4df320352b04eb09fa4f56e48516c9ae514decd0685775301c02a6f60bdb9dL332) had `digest_obj.algorithm.digest_size` instead, is there a difference here?

Reply:
See PR description / commit mesesage:
 
> Note that we now use the `padding.PSS.DIGEST_LENGTH` constant instead of passing the actual digest length, as we did before. Using the constant has the same result, but is recommended by the library documentation.

Reply:
I locally created two signature one with the constant and one with the digest length and can confirm that they do the same thing.

I also checked the code.

Reply:
https://github.com/pyca/cryptography/blob/25d794e079b44e6fac23791ca51da3a3fb9cb526/src/cryptography/hazmat/backends/openssl/rsa.py#L53-L54

Reply:
Right: I think `padding.PSS.DIGEST_LENGTH` is simply an internal `cryptography` constant that does look like it's used to end up using `digest_obj.algorithm.digest_size` in the end.

Reply:
Thanks! Oof my bad I didn't read the description first 🤦